lib/sassc/functions_handler.rb

```diff
@@ -10,25 +10,47 @@ def setup(native_options)
 
       list = Native.make_function_list(Script.custom_functions.count)
 
-      functs = FunctionWrapper.extend(Script::Functions)
-      functs.options = @options
+      functions = FunctionWrapper.extend(Script::Functions)
+      functions.options = @options
 
       Script.custom_functions.each_with_index do |custom_function, i|
-        @callbacks[custom_function] = FFI::Function.new(:pointer, [:pointer, :pointer]) do |s_args, cookie|
-          length = Native.list_get_length(s_args)
+        @callbacks[custom_function] = FFI::Function.new(:pointer, [:pointer, :pointer]) do |native_argument_list, cookie|
+          native_argument_list_length = Native.list_get_length(native_argument_list)
+          custom_function_arguments = []
 
-          v = Native.list_get_value(s_args, 0)
-          v = Native.string_get_value(v).dup
+          (0...native_argument_list_length).each do |i|
+            native_value = Native.list_get_value(native_argument_list, i)
 
-          s = Script::String.new(Script::String.unquote(v), Script::String.type(v))
+            case Native.value_get_tag(native_value)
+            when :sass_string
+              native_string = Native.string_get_value(native_value)
+              argument = Script::String.new(Script::String.unquote(native_string), Script::String.type(native_string))
 
-          value = functs.send(custom_function, s)
+              custom_function_arguments << argument
```

Comment:
it's unfortunate to have the sass dependency long term, but right at the moment, sass-rails requires sass too (because sprockets requires sass).  I suppose that would be a good solution for the present.

I can't exactly remember, but i think I may have had to tweak the String value class (i.e. it might not be exactly the same as it is in sass).